File: src/2766_Relocate_Marbles/solution.py
## https://leetcode.com/problems/relocate-marbles/

from typing import List


## TC: O(NlogN)
## SC: O(N)
def relocateMarbles(
    self, nums: List[int], moveFrom: List[int], moveTo: List[int]
) -> List[int]:
    s = set(nums)
    for f, t in zip(moveFrom, moveTo):
        s.remove(f)     ## O(1)
        if t not in s:  ## O(1)
            s.add(t)    ## O(1)

    return sorted(list(s))

File: src/2766_Relocate_Marbles/solution.ts
// https://leetcode.com/problems/relocate-marbles/

// TC: O(NlogN)
// SC: O(N)
function relocateMarbles(
  nums: number[],
  moveFrom: number[],
  moveTo: number[]
): number[] {
  let numsSet = new Set<number>(nums);
  let moveN = moveFrom.length;
  for (let i = 0; i < moveN; i++) {
    let f = moveFrom[i];
    let t = moveTo[i];
    numsSet.delete(f);
    if (!numsSet.has(t)) {
      numsSet.add(t);
    }
  }
  let numsArr = Array.from(numsSet).sort((n1, n2) => n1 - n2);
  return numsArr;
}
